source/extensions/tracers/opentelemetry/grpc_trace_exporter.cc

```diff
@@ -9,11 +11,38 @@ namespace OpenTelemetry {
 
 OpenTelemetryGrpcTraceExporter::OpenTelemetryGrpcTraceExporter(
     const Grpc::RawAsyncClientSharedPtr& client)
-    : client_(client, *Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
-                          "opentelemetry.proto.collector.trace.v1.TraceService.Export")) {}
+    : client_(client),
+      service_method_(*Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
+          "opentelemetry.proto.collector.trace.v1.TraceService.Export")) {}
+
+void OpenTelemetryGrpcTraceExporter::onCreateInitialMetadata(Http::RequestHeaderMap& metadata) {
+  metadata.setReferenceUserAgent(OtlpUtils::getOtlpUserAgentHeader());
+}
+
+void OpenTelemetryGrpcTraceExporter::onSuccess(
+    Grpc::ResponsePtr<ExportTraceServiceResponse>&& response, Tracing::Span&) {
+  if (response->has_partial_success()) {
+    auto msg = response->partial_success().error_message();
+    auto rejected_spans = response->partial_success().rejected_spans();
+    if (rejected_spans > 0 || !msg.empty()) {
+      if (msg.empty()) {
+        msg = "empty message";
+      }
+      ENVOY_LOG(warn, "OTLP partial success: {} ({} spans rejected)", msg, rejected_spans);
```

Comment:
This could be spammy at warn level. If the intent is to provide visibility to the server dropping spans, I suggest adding a counter.

Reply:
Would this be better at debug or info level?

Following the [go implementation](https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/otlptracegrpc/client.go#L177) and the [otel spec](https://opentelemetry.io/docs/specs/otlp/#partial-success), it seems that the message is intended to be a human readable error, so it should probably be logged at some level I think.

Reply:
Changed this to log at debug level instead, which aligns with the http exporter.